coordinator/src/node.rs

```diff
@@ -110,35 +111,25 @@ impl Node {
             "Closing position"
         );
 
-        let margin_trader = margin_trader(trade_params);
-        let margin_coordinator = margin_coordinator(trade_params);
-
         let leverage_long = leverage_long(trade_params);
         let leverage_short = leverage_short(trade_params);
 
-        let total_collateral = margin_coordinator + margin_trader;
+        let opening_price = trade_params.average_execution_price();
 
-        // FIXME: This is wrong as we cannot use the closing price to
-        // rebuild the payout function. We must save the initial price
-        // when creating the position and use it here again for
-        // closing.
-        let initial_price = trade_params.average_execution_price();
+        // FIXME: This is wrong as we cannot use the closing price to calculated the
+        // `accept_settlement_amount`. We must save the initial price when creating the position and
+        // use it here again for closing.
+        let closing_price = opening_price;
```

Comment:
🛠️ We have that in the position, it's the `average_entry_price` - because the position can be filled by multiple orders with different prices; there is no `opening_price`, but multiple of them, so we called it `average_entry_price`.

Reply:
but we don't store the position on the coordinator yet. so we can't get this price during closing yet.

Reply:
hm... that's unfortunate - can we just store them in memory in a hashmap for now as a quick fix? If that's too much work feel free to merge as is :)

Reply:
I'll merge as is and address your comment in a follow up PR.